kubetest2-gke/deployer/firewall.go

```diff
@@ -99,9 +134,9 @@ func (d *deployer) cleanupNetworkFirewalls(hostProject, network string) (int, er
 			return 0, fmt.Errorf("error deleting firewall: %v", errFirewall)
 		}
 		// It looks sometimes gcloud exits before the firewall rules are actually deleted,
-		// so sleep 10 seconds to wait for the firewall rules being deleted completely.
+		// so sleep 30 seconds to wait for the firewall rules being deleted completely.
 		// TODO(chizhg): change to a more reliable way to check if they are deleted or not.
-		time.Sleep(10 * time.Second)
+		time.Sleep(30 * time.Second)
```

Comment:
hmm, this is starting to become unreliable now. Do we use the firewall-rules immediately after anywhere to need to wait till we ensure it's actually deleted?

I think we should just return as soon as gcloud returns (even if it's running some hidden operation at the back that actually deletes asynchronously).

from kubetest2 perspective this is a best-effort cleanup and if gcloud says it's done deleting (by returning an exit code) then we shouldn't be waiting.

Reply:
Yeah it is pretty annoying, if I don't add the sleep (or sleep shorter like 10 sec), there will be an error in the next steps of deleting the subnetwork, the error message is like `ERROR: (gcloud.compute.networks.delete) Could not fetch resource:
 - The network resource 'projects/test-project-chizhg/global/networks/test-network5' is already being used by 'projects/test-project-chizhg/global/firewalls/test-network5-odm2l5xrgkhkgaqqdwsg4qbt'`. 

It could be a bug of `gcloud` but adding the hacky sleep here is the only easy way I can find to make it working.